Source/AwsCommonRuntimeKit/http/HTTPRequestOptions.swift

```diff
@@ -3,22 +3,26 @@
 import Foundation
 
 public struct HTTPRequestOptions {
-    public typealias OnIncomingHeaders = (_ stream: HTTPStream,
-                                          _ headerBlock: HTTPHeaderBlock,
+
+    public typealias OnInterimResponse = (_ statusCode: Int32,
                                           _ headers: [HTTPHeader]) -> Void
-    public typealias OnIncomingHeadersBlockDone = (_ stream: HTTPStream,
-                                                   _ headerBlock: HTTPHeaderBlock) -> Void
-    public typealias OnIncomingBody = (_ stream: HTTPStream, _ bodyChunk: Data) -> Void
-    public typealias OnStreamComplete = (_ stream: HTTPStream, _ error: CRTError?) -> Void
+    public typealias OnResponse = (_ statusCode: Int32,
+                                   _ headers: [HTTPHeader]) -> Void
+    public typealias OnTrailer = (_ headers: [HTTPHeader]) -> Void
+    public typealias OnIncomingBody = (_ bodyChunk: Data) -> Void
+    public typealias OnStreamComplete = (_ result: Result<Int32, CommonRunTimeError>) -> Void
 
     /// Outgoing request.
     let request: HTTPRequestBase
 
-    /// Invoked repeatedly as headers are received.
-    public let onIncomingHeaders: OnIncomingHeaders
+    /// Invoked when informational 1xx interim response is received
+    public let onInterimResponse: OnInterimResponse?
+
+    /// Invoked when main response headers are received.
+    public let onResponse: OnResponse
 
-    /// Invoked when response header block has been completely read.
-    public let onIncomingHeadersBlockDone: OnIncomingHeadersBlockDone
+    /// Invoked when trailer response is received.
+    public let onTrailer: OnTrailer?
 
     /// Invoked repeatedly as body data is received.
     public let onIncomingBody: OnIncomingBody
```

Comment:
again, declare OnIncomingBody before OnTrailer. just because that's the order these events occur